kernel/arch/dreamcast/fs/fs_dcload.c

```diff
@@ -146,21 +147,26 @@ void *dcload_open(vfs_handler_t * vfs, const char *fn, int mode) {
             return (void *)NULL;
         }
 
-        entry->hnd = hnd;
-        LIST_INSERT_HEAD(&dir_head, entry, fhlist);
-
         if(dcload_path)
             free(dcload_path);
```

Comment:
You can change this one too while you're at it - no need to NULL-check the pointer, free(NULL) is properly defined.

Reply:
Nevermind, I see you remove it in a following commit.